src/main/java/by/itech/lab/supplier/repository/CustomerRepository.java

```diff
@@ -12,5 +13,12 @@
     @Query("select c from Customer c where :active is null or c.status=:active")
     Page<Customer> findByStatus(Pageable pageable, @Param("active") Boolean active);
 
-}
+    @Modifying
+    @Query("update Customer set status = false where id = :id")
```

Comment:
need discuss

Reply:
whats result of the discussion?

Reply:
I suggest renaming "activate" method in BaseServise to "switchStatus" 
and implement method like this

`    public void switchStatus(final Long id) {
        final Customer existing = customerRepository
                .findById(id)
                .orElseThrow();
        if (existing.isStatus()) {
            customerRepository.deactivate(id);
        } else {
            customerRepository.activate(id);
        }`

Reply:
![image](https://user-images.githubusercontent.com/44436165/97689713-0cb84780-1aad-11eb-9be6-9d118ae99ebd.png)